### PR TITLE
Docker marketplace update

### PR DESCRIPTION
**Pull request by benbierens**
#### Infra
- Automated Docker builds for x86/arm
- Docker Compose example

### CODE REVIEW

File: docker/codex.Dockerfile [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
+RUN echo 'export NIMBLE_DIR="${HOME}/.nimble"' >> "${HOME}/.bash_env"
+RUN echo 'export PATH="${NIMBLE_DIR}/bin:${PATH}"' >> "${HOME}/.bash_env"
+
 WORKDIR /src
-RUN apk update && apk add git cmake curl make git bash linux-headers
 COPY . .
 RUN make clean
-RUN make update
-RUN make NIM_PARAMS="-d:disableMarchNative"
+RUN make -j4 update
```
Comment (dryajov) [inline docker/codex.Dockerfile:9]: we want this to be `-j` to be parametrizable, probably thru env variables.

Comment (veaceslavdoina) [reply docker/codex.Dockerfile]: Default value is 4 and we can adjust it using `build-arg`
```bash
docker build \
  --progress plain \
  --build-arg MAKE_PARALLEL=8 \
  --file docker/codex.Dockerfile .
```

```
#12 [builder 8/9] RUN make -j 8 update
#12 0.355 Creating nimbus-build-system.paths
```

File: codex/rest/api.nim [hunk 2]
```diff
@@ -45,6 +46,33 @@ proc validate(
...
+  let jarray = newJArray()
+  for key, value in peers:
+    jarray.add(%*
+      {
+        "peerId": $key,
+        "numberOfAddresses": $value.len
+      })
+
+  return jarray
+
```
Comment (benbierens) [inline codex/rest/api.nim:75]: formatEnginePeers and formatSwitchPeers can be remove, as well as their use in the JSON response below.

**Review by benbierens — COMMENTED** [review]
Small comment, otherwise I think this is looking pretty good.

**Comment by dryajov** [comment]
Can we resolve conflicts and add a description of whats being done here?

**Comment by dryajov** [comment]
there is a lot more going on here than just:
> Automated Docker builds for x86/arm
> Docker Compose example

We need to split out the changes related to the proofs out into it's own pr, before we can move forward with this.

**Comment by veaceslavdoina** [comment]
> there is a lot more going on here than just:

I added just part I did and know what it is about.

@emizzle, @benbierens can you please assist with the rest?

**Comment by benbierens** [comment]
Warning!

This change-set has been split into:
https://github.com/codex-storage/nim-codex/pull/449 - Docker multiarch build update
and
https://github.com/codex-storage/nim-codex/pull/450 - Simulate proof failures

(and some debug stuff I added which is obsolete, so I discard those changes.)

I'm closing this PR. It should not be merged. Let's double-check these smaller change sets and merge what we need merged.

**Pull request closed without merging**